### ISSUE TITLE
Hummingbird is now Kitsu

### ISSUE THREAD

**Issue by Kcchouette** [z411/trackma#263]
![Kitsu](https://cloud.githubusercontent.com/assets/1078430/20766713/247bd244-b740-11e6-98a6-e4f4d72b9c40.png)
> It also brings along a new API following the JSON API spec with full manga support.

I haven't yet find the API documentation, but Taiga seems to have begin the change: https://github.com/erengy/taiga/issues/341

**Comment by z411** [comment]
Thanks for the heads up. Seriously, where's the documentation?

**Comment by Kcchouette** [comment]
Based on Taiga analyse:
* The header is now `application/vnd.api+json`
* https://hummingbird.me/api/v1/search/anime?query=aono -> https://kitsu.io/api/edge/anime?filter[text]=aono

They use this standard to write API URL: http://jsonapi.org/format/#fetching-resources

**Comment by wopian** [comment]
>Thanks for the heads up. Seriously, where's the documentation?

It'll be documented soon, it's currently still being worked on.

Edge is API with the latest (sometimes breaking) changes, there will also be a versioned API very soon (/api/17).

@z411 you can email @joshfabian to get an invite to the dev slack group where you'd get all the information needed until it's publicly documented.

**Comment by wopian** [comment]
Documentation is being worked on over at http://docs.kitsu17.apiary.io

**Comment by z411** [comment]
@wopian Thank you. I had e-mailed Josh this morning but I haven't gotten a reply yet.

**Comment by wopian** [comment]
Further information can be found [here](https://forums.hummingbird.me/t/kitsu-api-documentation/33367)

**Comment by z411** [comment]
Started work on this in branch `kitsu`.

**Comment by NikkyAI** [comment]
is this being worked on and if not could @z411 make a todo list of features that have to be added to the https://github.com/z411/trackma/tree/kitsu branch? i would like to fix my trackma asap and PR seems the fastest way to get what i want here

**Comment by z411** [comment]
If you're so fast you can check the code yourself.

**Comment by z411** [comment]
Implemented list fetch, update, delete, search.

Fetching is currently very slow, I could use the cache to request show details just once, like libvndb does.

**Comment by NikkyAI** [comment]
on kitsu i have currently the issue that progress is x/0 this breaks differently in qt and gtkui, but it does break.. i cannot ind how you set the show length in other places thouhg.. guess i will look at this when i am not as tired.. later

**Comment by z411** [comment]
It's fixed now.

**Comment by Kcchouette** [comment]
Thank you @z411.
I'm waiting the great day of when `kitsu` will be merge in the `master` ^^

**Comment by NikkyAI** [comment]
to use the latest version from the kitsu branch use `sudo pip install -U git+https://github.com/z411/trackma.git@kitsu`

**Comment by z411** [comment]
API is unstable for now and there are several changes going on in order to fix some issues; it's also not possible to register an application-wide API key yet, so I won't be merging into `master` until it's in a stable state. But please feel free to use the `kitsu` branch and report any issues.

**Comment by z411** [comment]
Merged `kitsu` into `master` as it's in a relatively usable state for the time being.

**Comment by Kcchouette** [comment]
You can close this issue now @z411 . Thank you!

**Comment by NikkyAI** [comment]
somethin i noticed and that seems to not go away by itself is that new shows have zero episodes.. so detection bugs out, when i mark it as watched it goes to 1/0 etc..

might be related to using the unstable revisions in beteen versions but it only happens for new shows and only after kitsu.. but i suspect my local database migth be currupted could be as well

**Comment by z411** [comment]
Zero episodes just means that the total amount of episodes is not known yet. Everything else should still work. Are you getting some kind of error?

**Comment by wopian** [comment]
Would it not be more suitable to use `?` in that situation then?

**Comment by z411** [comment]
Sure, but that's not an issue specific to Kitsu. But I'll make the change either way, yeah.

**Comment by NikkyAI** [comment]
i also have to note that i cannpt advance a show in trackma-qt, it throws a error at me.. while the gtk version does that
and in qt it shows that i have some files downloaded while that bar does noz appear on gtk

**Comment by z411** [comment]
Fixed.

**Comment by z411** [comment]
Marking as implemented; if any issue arises please create a new issue.
